Sources_v3/Query/Sorting.swift

```diff
@@ -79,21 +77,20 @@ extension ChannelListSortingKey {
     }()
     
     func sortDescriptor(isAscending: Bool) -> NSSortDescriptor? {
-        var stringKeyPath: KeyPath<ChannelDTO, String>?
+        var intKeyPath: KeyPath<ChannelDTO, Int16>?
         var dateKeyPath: KeyPath<ChannelDTO, Date>?
         var optionalDateKeyPath: KeyPath<ChannelDTO, Date?>?
         
         switch self {
-        case .default: dateKeyPath = \ChannelDTO.defaultSortingAt
-        case .cid: stringKeyPath = \ChannelDTO.cid
-        case .type: stringKeyPath = \ChannelDTO.typeRawValue
-        case .createdAt: dateKeyPath = \ChannelDTO.createdAt
-        case .deletedAt: optionalDateKeyPath = \ChannelDTO.deletedAt
-        case .lastMessageAt: optionalDateKeyPath = \ChannelDTO.lastMessageAt
+        case .default: dateKeyPath = \.defaultSortingAt
+        case .createdAt: dateKeyPath = \.createdAt
+        case .updatedAt: dateKeyPath = \.updatedAt
+        case .lastMessageAt: optionalDateKeyPath = \.lastMessageAt
+        case .memberCount: intKeyPath = \.memberCount
         case .custom: break
         }
         
-        if let keyPath = stringKeyPath {
+        if let keyPath = intKeyPath {
             return .init(keyPath: keyPath, ascending: isAscending)
         }
         
```

Comment:
Maybe you can use `PartialKeyPath` for this:
```swift
        var keyPath: PartialKeyPath<ChannelDTO>?
        
        switch self {
        case .default: keyPath = \ChannelDTO.defaultSortingAt
        case .createdAt: keyPath = \ChannelDTO.createdAt
        ...
        }
        
        if let keyPath = keyPath {
            return .init(keyPath: keyPath, ascending: isAscending)
        }
```

Reply:
Switched to string key init usage 🙂